src/SFA.DAS.Assessor.Functions/Functions/Ofqual/OfqualDataReader.cs

```diff
@@ -0,0 +1,122 @@
+        {
+            if (!await FileExists(filePath))
+            {
+                logger.LogError("Failed to find Organisations data file at {filePath}.", filePath);
+                return new List<OfqualOrganisation>();
+            }
+
+            logger.LogInformation("Organisations data file found at {filePath}. Reading.", filePath);
+
+            string organisationsData = await _blobFileTransferClient.DownloadFile(filePath);
+            using var stringReader = new StringReader(organisationsData);
+            using var csvReader = new CsvReader(stringReader, CultureInfo.InvariantCulture);
+
+            var records = new List<OfqualOrganisation>();
+            csvReader.Read();
+            csvReader.ReadHeader();
+
+            while (csvReader.Read())
+            {
+                var record = new OfqualOrganisation(
+                    csvReader.GetField<string>("Recognition Number"),
+                    csvReader.GetField<string>("Name"),
+                    csvReader.GetField<string>("Legal Name"),
+                    csvReader.GetField<string>("Acronym"),
+                    csvReader.GetField<string>("Email"),
+                    csvReader.GetField<string>("Website"),
+                    csvReader.GetField<string>("Head Office Address Line 1"),
+                    csvReader.GetField<string>("Head Office Address Line 2"),
+                    csvReader.GetField<string>("Head Office Address Town/City"),
+                    csvReader.GetField<string>("Head Office Address County"),
+                    csvReader.GetField<string>("Head Office Address Postcode"),
+                    csvReader.GetField<string>("Head Office Address County"),
+                    csvReader.GetField<string>("Head Office Address Telephone Number"),
+                    csvReader.GetField<string>("Ofqual Status"),
+                    csvReader.TryGetField<DateTime?>("Ofqual Recognised From", out var ofQualFromDate) ? ofQualFromDate : null,
+                    csvReader.TryGetField<DateTime?>("Ofqual Recognised To", out var ofQualToDate) ? ofQualToDate : null,
+                    csvReader.GetField<string>("CCEA Regulation Status"),
+                    csvReader.TryGetField<DateTime?>("CCEA Regulation Recognised From", out var cceaFromDate) ? cceaFromDate : null,
+                    csvReader.TryGetField<DateTime?>("CCEA Regulation Recognised To", out var cceaToDate) ? cceaToDate : null
+                );
+                records.Add(record);
+            }
+
+            return records;
+        }
+
+        [FunctionName(nameof(ReadQualificationsData))]
+        public async Task<IEnumerable<OfqualStandard>> ReadQualificationsData([ActivityTrigger] string filePath, ILogger logger)
+        {
+            if (!await FileExists(filePath))
+            {
+                logger.LogError("Failed to find Qualifications data file at {filePath}.", filePath);
+                return new List<OfqualStandard>();
```

Comment:
Same comment as above and clearing the staging

Reply:
Done